testing-modules/junit-5/src/test/java/com/baeldung/junit5/testinstance/TweetSerializerJUnit4UnitTest.java

```diff
@@ -0,0 +1,57 @@
+package com.baeldung.junit5.testinstance;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.CoreMatchers.notNullValue;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertTrue;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class TweetSerializerJUnit4UnitTest {
+
+    private static String largeContent;
+    private static String content;
+    private static String smallContent;
+
+    private static Tweet tweet;
+
+    @BeforeClass
+    public static void setUpFixture() throws IOException {
+        content = "Lorem ipsum dolor sit amet, consectetur adipiscing elit";
+        smallContent = "Lorem ipsum dolor";
+        largeContent = new String(Files.readAllBytes(Paths.get("src/test/resources/lorem-ipsum.txt")));
+        tweet = new Tweet();
+        tweet.setId("AX1346");
+    }
+
+    @Test
+    public void serializerThrowsExceptionWhenMessageIsTooLarge() throws IOException {
+        tweet.setContent(largeContent);
+
+        TweetException tweetException = assertThrows(TweetException.class, () -> new TweetSerializer(tweet).serialize());
+        assertTrue(tweetException.getMessage()
+            .contains("Tweet is too large"));
+    }
+
+    @Test
+    public void serializerThrowsExceptionWhenMessageIsTooSmall() throws IOException {
+        tweet.setContent(smallContent);
+
+        TweetException tweetException = assertThrows(TweetException.class, () -> new TweetSerializer(tweet).serialize());
```

Comment:
I'm conflicted here. `assertThrows` is a JUnit5 assertion, so this isn't really a JUnit4 test you could use with JUnit4... obviously you _can_ use it with JUnit4 in the _JUnit5 vintage engine_. Can we modify this to make it more native JUnit4?